docs/design/deviceshifu/design-deviceShifu-plc4x.md

```diff
@@ -0,0 +1,77 @@
+# DeviceShifu PLC4X Design
+
+DeviceShifu-PLC4X allows shifu utilize [apache plc4x](https://plc4x.apache.org/) project to integrate PLC devices with various protocols such as S7, Modbus, etc.
+
+## Goal
+Create a deviceShifu-PLC4X type using PLC4X library and allow user to connect PLC devices easily
```

Comment:
what's the value of this goal? or what's difference we can provide after implement this feature?
maybe we can make it more explicitly about the word of "easy"?